sample: bluetooth: Document Peer Manager section of samples (#823)
nrfconnect/sdk-nrf-bm: samples/bluetooth/ble_hids_keyboard/README.rst

+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+
|                            | Input Report                | Read/Write/Notify | Encryption, no authentication | Connection must be encrypted to send key press and release events to the host.                                               |
+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+
|                            | Output Report               | Read/Write        | Encryption, no authentication | Connection must be encrypted for the host to read or set the LED state (Caps Lock, Num Lock, Scroll Lock).                   |
+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+

> |                            | Feature Report              | Read/Write        | Encryption, no authentication | Connection must be encrypted to exchange vendor-specific configuration data outside the normal input and output reports.     |

Reviewer: ```suggestion
|                            | Feature Report              | Read/Write        | Encryption, no authentication | Connection must be encrypted to exchange vendor-specific configuration data.                                                 |
```

LED 2:
   Lit when Caps Lock is active on the computer.

Security
********

The sample integrates :ref:`lib_peer_manager` (backed by :ref:`lib_bm_zms` for persistent storage) to support pairing and bonding.

The following security parameters are configured:

.. list-table:: Peer Manager security parameters
   :header-rows: 1

   * - Parameter
     - Value
     - Effect
   * - Bonding
     - Enabled
     - Supports storing of pairing information (bonding data).
   * - LE Secure Connections
     - Enabled
     - Supports LESC pairing.
   * - MITM protection
     - Disabled
     - Man-in-the-middle protection not required.
   * - Keypress notifications
     - Enabled

>      - Keypress events are sent to the peer during passkey entry.

Reviewer: ```suggestion
   * - Keypress notifications
     - Disabled
     - No keypress events are sent during passkey entry.
```

Author: Good catch, although I removed the Keypress notification entry in security params as per your request here https://github.com/nrfconnect/sdk-nrf-bm/pull/823#discussion_r3802341998

   * - I/O capabilities
     - Display and Yes/No entry
     - Determines local I/O capability, actual pairing method depends on negotiation with peer.

This sample configures the following services and their characteristics with these specific operation modes and required security levels:

+----------------------------+-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+
| Service                    | Characteristic              | Operation         | Required security level       | Effect                                                                                                                       |
+============================+=============================+===================+===============================+==============================================================================================================================+

> | HID Service                | HID Information             | Read              | Encryption, no authentication | Connection must be encrypted to read the HID version and device capability flags.                                            |

Reviewer: Please keep the characteristics in the same order as in https://www.bluetooth.com/specifications/specs/html/?src=hids_v1-0_1765320236/HIDS_v1.0/out/en/index-en.html#:~:text=Table%C2%A02.1.%C2%A0HID%20Service%20Characteristics (same order that the characteristics appear in the attribute table of this sample.)

Author: fixed

+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+
|                            | Report Map                  | Read              | Encryption, no authentication | Connection must be encrypted to read the report format the host needs to parse keyboard reports.                             |
+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+
|                            | Protocol Mode               | Read/Write        | Encryption, no authentication | Connection must be encrypted to switch the keyboard between Boot Protocol Mode and Report Protocol Mode.                     |
+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+

> |                            | Boot Keyboard Input Report  | Read/Write/Notify | Encryption, no authentication | Connection must be encrypted to send key presses to hosts that only support the HID boot protocol, such as a BIOS.           |

Reviewer: ```suggestion
|                            | Boot Keyboard Input Report  | Read/Write/Notify | Encryption, no authentication | Connection must be encrypted to send key presses with this characteristic when in the boot protocol mode.           |
```
Thinking we should avoid mentioning BIOS here.

Author: Fixed here and in hid mouse

+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+

> |                            | Boot Keyboard Output Report | Read/Write        | Encryption, no authentication | Connection must be encrypted to receive LED state updates (Caps Lock, Num Lock, Scroll Lock) from boot protocol hosts.       |

Reviewer: ```suggestion
|                            | Boot Keyboard Output Report | Read/Write        | Encryption, no authentication | Connection must be encrypted for the host to read or update the keyboard indicator LED state (Caps Lock, Num Lock, Scroll Lock).                |
```
Thinking this should be written from the perspective of a controller/client.

Author: Ok, fixed

+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+
|                            | Input Report                | Read/Write/Notify | Encryption, no authentication | Connection must be encrypted to send key press and release events to the host.                                               |
+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+

> |                            | Output Report               | Read/Write        | Encryption, no authentication | Connection must be encrypted to receive LED state updates (Caps Lock, Num Lock, Scroll Lock) from the host.                  |

Reviewer: ```suggestion
|                            | Output Report               | Read/Write        | Encryption, no authentication | Connection must be encrypted for the host to read or update the keyboard indicator LED state (Caps Lock, Num Lock, Scroll Lock).                  |
```

Author: Ok, fixed

+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+

> |                            | Feature Report              | Read/Write        | Encryption, no authentication | Connection must be encrypted to exchange vendor-specific configuration data outside the normal input and output reports.     |

Reviewer: ```suggestion
|                            | Feature Report              | Read/Write        | Encryption, no authentication | Connection must be encrypted for a host to read or write the Feature Report.     |
```

Author: I would prefer to keep it as is, the suggestion doesn't really add anything, it just restates what's already conveyed by the Characteristic/Operation/Security fields, instead of explaining what the actual effect is.

Reviewer 2: I think the suggestion from @anhmolt makes sense. Why mix in the input and output reports here? It makes it more confusing in my opinion.

Reviewer: Yes, I think mentioning the other report characteristics here does not really make sense. I was also a bit unsure on mentioning 'vendor-specific' here, but reading some more, I think it's fine.

+                            +-----------------------------+-------------------+-------------------------------+------------------------------------------------------------------------------------------------------------------------------+

> |                            | Control Point               | Write             | Encryption, no authentication | Connection must be encrypted to suspend or resume HID report notifications, for example when the host enters low power mode. |

Reviewer: ```suggestion
|                            | HID Control Point               | Write             | Encryption, no authentication | Connection must be encrypted to suspend or resume HID report notifications, for example when the host enters low power mode. |
```

Author: fixed
